applicationset/controllers/applicationset_controller.go

```diff
@@ -134,6 +138,45 @@ func (r *ApplicationSetReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{RequeueAfter: ReconcileRequeueOnValidationError}, nil
 	}
 
+	applications, err := r.getCurrentApplications(ctx, applicationSetInfo)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to get current applications for application set: %w", err)
+	}
+
+	appSyncMap := map[string]bool{}
+	appMap := map[string]argov1alpha1.Application{}
+
+	if r.EnableProgressiveRollouts && applicationSetInfo.Spec.Strategy != nil {
```

Comment:
Should we do the same `strategy != nil && strategy.Type == "RollingSync"` here as below? Maybe make it a method?